Integration/UrlStripperIntegration.php

```diff
@@ -0,0 +1,162 @@
+<?php
+
+/*
+ * @copyright   2018 Mautic Contributors. All rights reserved
+ * @author      Digital Media Solutions, LLC
+ *
+ * @link        http://mautic.org
+ *
+ * @license     GNU/GPLv3 http://www.gnu.org/licenses/gpl-3.0.html
+ */
+
+namespace MauticPlugin\MauticEnhancerBundle\Integration;
+
+use Mautic\CoreBundle\Form\Type\YesNoButtonGroupType;
+use Mautic\LeadBundle\Entity\Lead;
+
+/**
+ * Class UrlStripperIntegration.
+ *
+ * Allow verification of a lead's email address using X-verify on a configurable
+ * list of campaigns.
+ *
+ * @todo Hook up to the plugin stats
+ */
+class UrlStripperIntegration extends AbstractEnhancerIntegration
+{
+    /**
+     * @return string
+     */
+    public function getAuthenticationType()
+    {
+        return 'none';
+    }
+
+    /**
+     * @return string
+     */
+    public function getName()
+    {
+        return 'UrlStripper';
+    }
+
+    /**
+     * @return string
+     */
+    public function getDisplayName()
+    {
+        return 'Normalize Consent URL with '.$this->getName();
+    }
+
+    /**
+     * @return array|mixed
+     */
+    protected function getEnhancerFieldArray()
+    {
+        return [
+            'consent_url_clean' => [
+                'label' => 'Consent URL - Cleaned',
+                'type'  => 'text',
+            ],
+        ];
+    }
+
+    /**
+     * @return array
+     */
+    public function getSupportedFeatures()
+    {
+        return ['push_lead'];
+    }
+
+    /**
+     * @param Lead $lead
+     *
+     * @return bool
+     */
+    public function doEnhancement(Lead $lead)
+    {
+        $persist = false;
+        if (!empty($lead)) {
+            $settings                = $this->getIntegrationSettings()->getFeatureSettings();
+            $originalConsentUrlField = $settings['original_consent_url'];
+
+            try {
+                $fieldValue = $lead->getFieldValue($originalConsentUrlField);
+                if (!empty($fieldValue)) {
+                    // remove superfilious params
+                    $cleanFieldValue = strtok($fieldValue, '?');
+                    $lead->addUpdatedField('consent_url_clean', $cleanFieldValue);
+                    $this->logger->addDebug(
+                        'CONSENT URL NORMALIZER: verification values to update: '.$fieldValue.' => '.$cleanFieldValue
+                    );
+                    $persist = true;
```

Comment:
Before returning true, could you check to make sure the original value of consent_url_clean is different than the new value? Just to keep consistent with other enhancers.

Reply:
check out this version.